crates/db/migrations/20251216142123_refactor_task_attempts_to_workspaces_sessions.sql

```diff
@@ -0,0 +1,121 @@
+DROP INDEX IF EXISTS idx_execution_processes_task_attempt_type_created;
+
+CREATE TABLE execution_processes_new (
+    id              BLOB PRIMARY KEY,
+    session_id      BLOB NOT NULL,
+    run_reason      TEXT NOT NULL DEFAULT 'setupscript'
+                       CHECK (run_reason IN ('setupscript','codingagent','devserver','cleanupscript')),
+    executor_action TEXT NOT NULL DEFAULT '{}',
+    status          TEXT NOT NULL DEFAULT 'running'
+                       CHECK (status IN ('running','completed','failed','killed')),
+    exit_code       INTEGER,
+    dropped         INTEGER NOT NULL DEFAULT 0,
+    started_at      TEXT NOT NULL DEFAULT (datetime('now', 'subsec')),
+    completed_at    TEXT,
+    created_at      TEXT NOT NULL DEFAULT (datetime('now', 'subsec')),
+    updated_at      TEXT NOT NULL DEFAULT (datetime('now', 'subsec')),
+    FOREIGN KEY (session_id) REFERENCES sessions(id) ON DELETE CASCADE
+);
+
+-- Since we used workspace.id as session.id, the task_attempt_id values map directly
+INSERT INTO execution_processes_new (id, session_id, run_reason, executor_action, status, exit_code, dropped, started_at, completed_at, created_at, updated_at)
+SELECT id, task_attempt_id, run_reason, executor_action, status, exit_code, dropped, started_at, completed_at, created_at, updated_at
+FROM execution_processes;
+
+DROP TABLE execution_processes;
+ALTER TABLE execution_processes_new RENAME TO execution_processes;
+
+-- Recreate execution_processes indexes
+CREATE INDEX idx_execution_processes_session_id ON execution_processes(session_id);
+CREATE INDEX idx_execution_processes_status ON execution_processes(status);
+CREATE INDEX idx_execution_processes_run_reason ON execution_processes(run_reason);
+
+-- 8. Rename executor_sessions to coding_agent_turns and drop task_attempt_id
+-- (needs rebuild to drop the redundant task_attempt_id column)
+CREATE TABLE coding_agent_turns (
+    id                    BLOB PRIMARY KEY,
+    execution_process_id  BLOB NOT NULL,
+    session_id            TEXT,
+    prompt                TEXT,
+    summary               TEXT,
+    created_at            TEXT NOT NULL DEFAULT (datetime('now', 'subsec')),
+    updated_at            TEXT NOT NULL DEFAULT (datetime('now', 'subsec')),
+    FOREIGN KEY (execution_process_id) REFERENCES execution_processes(id) ON DELETE CASCADE
+);
+
+INSERT INTO coding_agent_turns (id, execution_process_id, session_id, prompt, summary, created_at, updated_at)
+SELECT id, execution_process_id, session_id, prompt, summary, created_at, updated_at
+FROM executor_sessions;
+
+DROP TABLE executor_sessions;
+
+-- Recreate coding_agent_turns indexes
+CREATE INDEX idx_coding_agent_turns_execution_process_id ON coding_agent_turns(execution_process_id);
+CREATE INDEX idx_coding_agent_turns_session_id ON coding_agent_turns(session_id);
+
+-- 9. attempt_repos: no changes needed - FK auto-updated when task_attempts renamed to workspaces
```

Comment:
Why not rename `attempt_repos` to `workspace_repos` here now that `attempt` is a legacy concept?